ast/json.go

```diff
@@ -1051,6 +1053,10 @@ func unmarshalExpression(msg json.RawMessage) (Expression, error) {
 	if err != nil {
 		return nil, err
 	}
+	if n == nil {
+		// If both n and err are nil, then this is a BadExpression, just return nil.
+		return nil, nil
```

Comment:
The short answer is when you get here `err` is already `nil`.  If `err` is not `nil`, the function would return at line 1054

Longer answer:
In Go, if we failed to parse something, like the operand expression, we just assign it nil and the checker will bring that up in a later phase. In Rust, we cannot do the same thing so we used the `BadExpression` to wrap it up instead of assigning it nil.

The err here should be unmarshall error, not the semantic error. so we won't be returning an error here.